mmdeploy/codebase/mmdet3d/deploy/mono_detection.py

```diff
@@ -0,0 +1,207 @@
+            'test_dataloader', 'val_dataloader', 'train_dataloader'
+    ]:
+        if dataloader_name not in model_cfg:
+            continue
+        dataloader_cfg = model_cfg[dataloader_name]
+        dataset_cfg = dataloader_cfg.dataset
+        if 'metainfo' in dataset_cfg:
+            return dataset_cfg.metainfo
+    return None
+
+
+@MMDET3D_TASK.register_module(Task.MONO_DETECTION.value)
+class MonoDetection(BaseTask):
+
+    def __init__(self, model_cfg: mmengine.Config, deploy_cfg: mmengine.Config,
+                 device: str):
+        super().__init__(model_cfg, deploy_cfg, device)
+
+    def build_backend_model(self,
+                            model_files: Sequence[str] = None,
+                            **kwargs) -> torch.nn.Module:
+        """Initialize backend model.
+
+        Args:
+            model_files (Sequence[str]): Input model files.
+
+        Returns:
+            nn.Module: An initialized backend model.
+        """
+        from .mono_detection_model import build_mono_detection_model
+
+        data_preprocessor = deepcopy(
+            self.model_cfg.model.get('data_preprocessor', {}))
+        data_preprocessor.setdefault('type', 'mmdet3D.Det3DDataPreprocessor')
+
+        model = build_mono_detection_model(
+            model_files,
+            self.model_cfg,
+            self.deploy_cfg,
+            device=self.device,
+            data_preprocessor=data_preprocessor)
+        model = model.to(self.device)
+        return model
+
+    def create_input(
+        self,
+        pcd: Union[str, Sequence[str]],
+        input_shape: Sequence[int] = None,
+        data_preprocessor: Optional[BaseDataPreprocessor] = None
+    ) -> Tuple[Dict, torch.Tensor]:
+        """Create input for detector.
+
+        Args:
+            pcd (str, Sequence[str]): Input pcd file path.
+            input_shape (Sequence[int], optional): model input shape.
+                Defaults to None.
+            data_preprocessor (Optional[BaseDataPreprocessor], optional):
+                model input preprocess. Defaults to None.
+
+        Returns:
+            tuple: (data, input), meta information for the input pcd
+                and model input.
+        """
+        cfg = self.model_cfg
+        test_pipeline = deepcopy(cfg.test_dataloader.dataset.pipeline)
+        test_pipeline = Compose(test_pipeline)
+        box_type_3d, box_mode_3d = \
+            get_box_type(cfg.test_dataloader.dataset.box_type_3d)
+
+        if isinstance(pcd, (list, tuple)):
+            pcd = pcd[0]
+        path_prefix = os.path.dirname(pcd)
+        data_list = mmengine.load(pcd)['data_list']
+        assert 1 == len(data_list)
+        assert 'images' in data_list[0]
+
+        data = []
+        images = data_list[0]['images']
+        for cam_type, item in images.items():
+            # avoid data_info['images'] has multiple keys anout camera views.
+            # Only support one cam type here
+            if cam_type != 'CAM_BACK':
```

Comment:
@zhjunqin Hi, could change to the following code and it works with kitti dataset.
```
            if not cam_type in ['CAM_FRONT', 'CAM2']:
                continue
```

Reply:
Done.